tests/phpunit/CiviTest/CiviTestSMSProvider.php

```diff
@@ -44,10 +44,10 @@ public static function &singleton($providerParams = [], $force = FALSE) {
     return self::$_singleton[$cacheKey];
   }
 
-  public function send($recipients, $header, $message, $dncID = NULL) {
+  public function send($recipients, $header, $message, $jobID = NULL, $userID = NULL) {
     $this->sentMessage = $message;
     $sid = substr(str_shuffle('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'), 0, 16);
-    $this->createActivity($sid, $message, $header, $dncID);
+    $this->createActivity($sid, $message, $header, $jobID, $userID);
```

Comment:
My thinking for the fail is that now that you've added the $userID (contact id) here it is correctly creating the additional SMS delivery activity, whereas before it couldn't figure out the contact id without that and so skipped it. So I think the test is sort of "wrong" because it should be expecting both the SMS activity and the SMS delivery to be there too.